### DISCUSSION TITLE
On mobile when calling hook/action to write to contract mobile browser won't change to Metamask to approve the tx

### DISCUSSION THREAD

**Discussion by kobyle69** [wevm/wagmi#1865] [Q&A]
Hello,

On PC everything works just fine.

But on mobile when calling hook/action to write to contract mobile browser won't change to Metamask to approve the tx.

Using React, Wagmi and ConnectKit.

**Comment by leeburn** [comment]
Have you solved this yet?

I was having the same issue with WalletConnect Legacy on Wagmi 0.6.8, so I upgraded to Wagmi 0.12.0 and WalletConnect 2 and now MetaMask is missing from WalletConnect entirely.
